main.c

```diff
@@ -137,8 +146,8 @@ static void linux_boot(const struct boot_params boot_params)
     die_on(LINUX_HEADER_OFFSET + linux_header_size > sizeof(*linux_params),
            "Invalid linux header size");
 
-    memcpy((char *)linux_params + LINUX_HEADER_OFFSET,
-           (const char *)(boot_params.kernel_addr + LINUX_HEADER_OFFSET),
+    memcpy((uint8_t *)linux_params + LINUX_HEADER_OFFSET,
+           (const uint8_t *)(boot_params.kernel_addr + LINUX_HEADER_OFFSET),
```

Comment:
This is going to change in the follow up MR. No real need to change the type here. Ideally this would at least be an extra commit that also explains *why* this is better. But I'd rather you drop the change from this commit and just fix it during the follow up.

Reply:
kk